packages/rich-text/src/register-custom-alignment-type.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * WordPress dependencies
+ */
+import { dispatch } from '@wordpress/data';
+
+export function registerCustomAlignmentType( name, settings ) {
```

Comment:
I know that this is a draft PR but I'd love to stress how valuable some inline documentation right here can be. people will look here to try and find the answer "what information is necessary to define an alignment type?" (if only we had some way to designate _types_ and have the computer answer the question for us…)

looking through the whole PR I'm able to discern that it needs a `name`, a `type`, and `control` but I still don't know what those properties should be.

we could consider adding a [JSDoc typedef](http://usejsdoc.org/tags-typedef.html)
```js
/**
 * @typedef {Object} AlignmentType
 * @property {string} type - blockType for which this will be available
 * @property {AlignmentControl} control - whatever this actually is, maybe another typedef
 */

/**
 * Registers a custom alignment type
 *
 * @example
 * registerCustomAlignmentType( 'justified', { type: 'core/paragraph', control: <Whatever /> } );
 *
 * @param {string} name - name of alignment, e.g. "left" or "justified"
 * @param {AlignmentType} settings - specifies the alignment
 */
export function registerCustomAlignmentType( name, settings ) {
  …
}
```

even without the typedef though having an example use can clear up so much confusion, as can enumerating what properties of `settings` may exist and what they mean

Reply:
Thanks for pointing this out @dmsnell!

That file is a very stripped down version of [register-format-type.js](https://github.com/WordPress/gutenberg/blob/37bc73524d8756b5ebd0b5602a6b0b1580e4025f/packages/rich-text/src/register-format-type.js).
It doesn't have any docs, validations, etc, because this exploration is more about extending the toolbar than handling this registration.
I will add everything (and more!) back as soon as I get the extension part right. 👍